runtime/compiler/control/CompilationThread.cpp

```diff
@@ -11658,6 +11658,16 @@ TR::CompilationInfoPerThreadBase::processExceptionCommonTasks(
                                  _methodBeingCompiled->getMethodDetails().getMethod(),
                                  translationTime,
                                  compilationErrorNames[_methodBeingCompiled->_compErrCode]);
+            uintptr_t vmState = vmThread->omrVMThread->vmState;
+            TR_VerboseLog::write(" VmState=0x%08.8x", (unsigned int)vmState);
+            if (_methodBeingCompiled->_compErrCode == compilationHeapLimitExceeded
+                && (vmState >> 16) == 0x0005
```

Comment:
Looking at the decoding login in https://github.com/eclipse-openj9/openj9/blob/master/runtime/compiler/control/J9Options.cpp#L651 I see the following:
vmstate = 0xaabbcc
1. aa needs to be 0x5
2. if bb==00 ==> This is ilgen
3. if bb==ff ==> this is codegen and cc is the codegen phase (note that 0x5ffff case is not covered) 
4. else bb gives us the optimization and cc is the analysis phase

Based on this I propose we use a test like:
```
if (_methodBeingCompiled->_compErrCode == compilationHeapLimitExceeded && (vmState & J9VMSTATE_JIT) == J9VMSTATE_JIT)
{
  // Test that we failed during an optimization.
  uint8_t jitphase = (vmState >> 8) & 0xff;
  if (jitPhase != 0 &&  // not ilgen
      jitPhase != 0xff) // not codegen and not before ilgen
     {
	 }
```